### PR TITLE
Fix SDL.OnAppPermissionConsent with 'externalConsentStatus' parameter

### PR DESCRIPTION
**Pull request by istoilovgithub**
The fix changes the permission change notifications to be send according
to the previous consent status state of the group which will be consented

### CODE REVIEW

**Comment by istoilovgithub** [comment]
@okozlovlux Please review

**Comment by istoilovgithub** [comment]
@VVeremjova @okozlovlux Fixed failed UTs with 9d9713d, please review

File: src/components/policy/policy_external/test/policy_manager_impl_test.cc [hunk 1]
```diff
@@ -714,7 +714,7 @@ TEST_F(
 
   EXPECT_TRUE(policy_manager_->GetCache()->ApplyUpdate(t));
 
-  EXPECT_CALL(listener_, OnPermissionsUpdated(app_id_1_, _)).Times(AtLeast(2));
+  EXPECT_CALL(listener_, OnPermissionsUpdated(app_id_1_, _)).Times(1);
```
Comment (okozlovlux) [inline src/components/policy/policy_external/test/policy_manager_impl_test.cc]: @istoilovgithub , `Times(1)` is redundant

Comment (istoilovgithub) [inline src/components/policy/policy_external/test/policy_manager_impl_test.cc]: @okozlovlux Agreed, fixed with ace66ca.

**Pull request merged**